File: 17solve/17-153.py
# Автор В.Н. Шубинкин

with open('17-1.txt') as f:
    a = [int(x) for x in f.readlines()]

count = 0
min_r = 20000
for i in range(len(a) - 1):
    if a[i + 1] > a[i]:
        count += 1
        min_r = min(min_r, a[i + 1] - a[i])
print(count, min_r)

File: 17solve/17-153.pas
// Автор В.Н. Шубинкин
##
var a := ReadLines('17-1.txt').Select(t -> t.ToInteger).ToArray;
var count := 0;
var minR := 20000;
for var i := 0 to a.High - 1 do
  if a[i] < a[i + 1] then
  begin
    count += 1;
    minR := min(minR, a[i + 1] - a[i])    
  end;
println(count, minR)